### PR TITLE
Emit lookup stat correctly

### PR DESCRIPTION
**Pull request by mennopruijssers**
Previously the lookup stat was emitted from the event-handler inside of `ringpop.go` when a `LookupEvent` was emitted. However, the lookup-event is emitted on ringpop itself so it's internal event-handler isn't triggered.

### CODE REVIEW

File: ringpop.go [hunk 1]
```diff
@@ -612,7 +612,20 @@ func (rp *Ringpop) LookupN(key string, n int) ([]string, error) {
...
+	duration := time.Now().Sub(startTime)
+	rp.statter.RecordTimer(rp.getStatKey(fmt.Sprintf("lookupn.%d", n)), nil, duration)
+
+	if len(destinations) == 0 {
+		err := errors.New("could not find destinations for key")
+		rp.logger.WithField("key", key).Warn(err)
+		return destinations, err
+	}
+
+	return destinations, nil
```
Comment (Motiejus) [inline ringpop.go]: Nice. Thank you for adding this.

File: ringpop.go [hunk 2]
```diff
@@ -612,7 +612,20 @@ func (rp *Ringpop) LookupN(key string, n int) ([]string, error) {
...
+
+	destinations := rp.ring.LookupN(key, n)
+
+	duration := time.Now().Sub(startTime)
+	rp.statter.RecordTimer(rp.getStatKey(fmt.Sprintf("lookupn.%d", n)), nil, duration)
+
+	if len(destinations) == 0 {
+		err := errors.New("could not find destinations for key")
+		rp.logger.WithField("key", key).Warn(err)
+		return destinations, err
```
Comment (Motiejus) [inline ringpop.go]: Could we somehow unit-test this branch too?

File: ringpop.go [hunk 3]
```diff
@@ -612,7 +612,20 @@ func (rp *Ringpop) LookupN(key string, n int) ([]string, error) {
 	if !rp.Ready() {
 		return nil, ErrNotBootstrapped
 	}
-	return rp.ring.LookupN(key, n), nil
+	startTime := time.Now()
```
Comment (btromanova) [inline ringpop.go]: Does it make sense to emit event from lookupN as well? We do it in Lookup.

**Comment by btromanova** [comment]
Just one question about event from LookupN. Otherwise, lgtm. I've double checked that there are no other events that are processed in Ringpop HandleEvent and emitted form Ringpop itself.

File: ringpop.go (hunk 3 above)
Comment (mennopruijssers) [inline ringpop.go]: good catch!

File: ringpop.go (hunk 2 above)
Comment (mennopruijssers) [inline ringpop.go]: Both method (lookup/lookupN) are poorly tested. I added tests for both of them. All lines/branches are covered now :)

File: ringpop_test.go [hunk 4]
```diff
@@ -577,6 +588,33 @@ func (s *RingpopTestSuite) TestLookupNNotReady() {
...
+	s.Error(err)
+}
+
+func (s *RingpopTestSuite) TestLookupN() {
+	createSingleNodeCluster(s.ringpop)
+
+	result, err := s.ringpop.LookupN("foo", 5)
+
+	s.Nil(err)
+	s.Equal(1, len(result), "LookupN returns not more results then number of nodes")
```
Comment (Motiejus) [inline ringpop_test.go]: s/then/than

File: ringpop.go [hunk 5]
```diff
@@ -619,6 +619,12 @@ func (rp *Ringpop) LookupN(key string, n int) ([]string, error) {
 	duration := time.Now().Sub(startTime)
 	rp.statter.RecordTimer(rp.getStatKey(fmt.Sprintf("lookupn.%d", n)), nil, duration)
 
+	rp.emit(events.LookupNEvent{
+		Key:      key,
+		N: n,
```
Comment (Motiejus) [inline ringpop.go]: Indentation seems to be off, did you run gofmt?

**Comment by Motiejus** [comment]
It's better to rebase on top of `origin/dev` rather than merge it to your branch. Could you do that?

File: events/events.go [hunk 6]
```diff
@@ -55,6 +55,13 @@ type LookupEvent struct {
 	Duration time.Duration
 }
 
+// A LookupNEvent is sent when a lookupN is performed on the Ringpop's ring
+type LookupNEvent struct {
+	Key      string
+	N 		 int
```
Comment (Motiejus) [inline events/events.go]: Indentation seems to be off.. `gofmt`?

**Comment by Motiejus** [comment]
LGTM

**Comment by btromanova** [comment]
lgtm

**Pull request merged**